harness/determined/profiler.py

```diff
@@ -297,81 +310,64 @@ def run(self) -> None:
-                self.current_batch.add_nongpu_measurement(SysMetricType.DISK_THRU_WRITE_METRIC, disk_write_thru_measurement)
-                self.current_batch.add_nongpu_measurement(SysMetricType.FREE_MEM_METRIC, free_memory_measurement)
-                self.current_batch.add_nongpu_measurement(SysMetricType.SIMPLE_CPU_UTIL_METRIC, cpu_util_measurement)
-
+                cpu_util = cpu_util_collector.measure(immutable_batch_idx)
+                gpu_util = gpu_util_collector.measure(immutable_batch_idx)
+                gpu_memory = gpu_memory_collection.measure(immutable_batch_idx)
+                net_thru_sent, net_thru_recv = network_throughput_collector.measure(immutable_batch_idx)
+                free_memory = free_memory_collector.measure(immutable_batch_idx)
+                disk_read_thru, disk_write_thru, iops = disk_collector.measure(immutable_batch_idx)
+
+                for gpu_uuid in gpu_util.keys():
+                    self.current_batch.add_gpu_measurement(SysMetricType.GPU_UTIL_METRIC, gpu_uuid, gpu_util[gpu_uuid])
+
+                for gpu_uuid in gpu_memory.keys():
+                    self.current_batch.add_gpu_measurement(SysMetricType.GPU_FREE_MEMORY_METRIC, gpu_uuid, gpu_util[gpu_uuid])
+
+                self.current_batch.add_nongpu_measurement(SysMetricType.NET_THRU_SENT_METRIC, net_thru_sent)
+                self.current_batch.add_nongpu_measurement(SysMetricType.NET_THRU_RECV_METRIC, net_thru_recv)
+                self.current_batch.add_nongpu_measurement(SysMetricType.DISK_IOPS_METRIC, iops)
+                self.current_batch.add_nongpu_measurement(SysMetricType.DISK_THRU_READ_METRIC, disk_read_thru)
+                self.current_batch.add_nongpu_measurement(SysMetricType.DISK_THRU_WRITE_METRIC, disk_write_thru)
+                self.current_batch.add_nongpu_measurement(SysMetricType.FREE_MEM_METRIC, free_memory)
+                self.current_batch.add_nongpu_measurement(SysMetricType.SIMPLE_CPU_UTIL_METRIC, cpu_util)
                 last_measurement_time = time.time()
 
-
             # Check if it is time to flush the batch and start a new batch
             if time.time() - batch_start_time > self.FLUSH_INTERVAL:
-                self.enqueue_for_async_send(self.current_batch.convert_to_post_format())
+                self.send_queue.put(self.current_batch.convert_to_post_format())
                 self.current_batch.clear()
                 batch_start_time = time.time()
 
-            time.sleep(0.02)
-
-
-
-
-
-
-
-
 
 # This is a thread that exists solely so that we can make API calls without blocking
 # It has a Queue through which work is sent to the thread
 class ProfilerSenderThread(threading.Thread):
-    POLL_INTERVAL_SECS = 0.5
     def __init__(self, inbound_queue: queue.Queue, master_url: str) -> None:
         self.master_url = master_url
         self.inbound_queue = inbound_queue
-        self.quitting = False
-        super().__init__()
+        self.control_queue = queue.Queue()
```

Comment:
where are we reading off this control queue, am I missing it?

Reply:
good catch, the inbound queue also serves as the control so we don't need that extra queue.